maze_transformer/generation/generators.py

```diff
@@ -94,6 +99,13 @@ def gen_dfs(
             ),
         )
 
+    @classmethod
+    def gen_dfs_with_solution(cls, grid_shape: Coord):
+        maze = cls.gen_dfs(grid_shape)
+        solution = np.array(maze.generate_random_path())
```

Comment:
Yep in DFS these are the same. But good point that this will become confusing when we add new maze gen algorithms. I'll add a comment about this to the relevant issue.

Reply:
`generate_random_path` uses the [A* algorithm](https://en.wikipedia.org/wiki/A*_search_algorithm), so if it's implemented correctly it will always find the shortest path from the specified source to the specified goal. The docstring for `generate_random_path` could be updated to clarify this.